samples/js/picasa/find-photos/src/main.js

```diff
@@ -204,6 +205,38 @@ async function getSizes(input: Object, origWidth: number, origHeight: number)
   return new Map(tuples);
 }
 
+async function getResources(input: Object, origWidth: number, origHeight: number)
+: Promise<Map<Struct, Struct>> {
+  const thumbURL = (await input.mediaQ24group.mediaQ24thumbnail
+    .get(0)).url.split('/');
+  const sizePart = thumbURL.length - 2;
+
+  const makeURL = s => {
+    thumbURL[sizePart] = `s${s}`;
+    return thumbURL.join('/');
+  };
+
+  const tuples = sizes.map(s => {
+    const r = fit(origWidth, origHeight, s, s);
+    if (r.scale > 1) {
+      // $FlowIssue: Does not understand that filter removes all null values.
+      return null;
+    }
+    return [
+      newStruct('', {width: r.width, height: r.height}),
+      newStruct('RemoteResource', {url: makeURL(s)}),
+    ];
+  }).filter(t => t);
+
+  // The original file.
+  tuples.push([
+    newStruct('', {width: origWidth, height: origHeight}),
+    newStruct('RemoteResource', {url: makeURL('d')}),
+  ]);
+
+  return new Map(tuples);
+}
```

Comment:
Do you mean getSizes(). If so, I thought about it and given that the return types in this code are different, it would be kind of hard to factor most of it out without resorting to Map<Struct, Any>. This might be worth it but my thinking was that getSizes was going to go away once the UI started using resources rather than sizes. Let me know if you disagree with that.